src/lib.rs

```diff
@@ -1493,4 +1553,24 @@ mod tests {
         assert_ne!(a, c);
         assert_ne!(a, d);
     }
+
+    #[test]
+    fn tensor_display() {
+        let tests = [
+                ("1", &[][..], &[1][..]),
+                ("[1]", &[1], &[1]),
+                ("[1, 2]", &[2], &[1, 2]),
+                ("[[1, 2], [3, 4]]", &[2, 2], &[1, 2, 3, 4]),
+                ("[[[1], [2]], [[3], [4]]]", &[2, 2, 1], &[1, 2, 3, 4]),
+                ("[[], []]", &[2, 0], &[]),
+                ("[[], []]", &[2, 0, 2], &[]),
```

Comment:
Is this is the least surprising format for these edge case shapes?

Reply:
I think so.  If you go to play.rust-lang.org and enter:
```rust
fn main() {
  println!("{:?}", <[[[i32; 2]; 0]; 2]>::default());
}
```
you get `[[], []]`, and it would probably be least surprising to be consistent with how arrays are formatted.